src/ray/core_worker/transport/direct_task_transport.cc

```diff
@@ -507,7 +507,23 @@ void CoreWorkerDirectTaskSubmitter::RequestNewWorkerIfNeeded(
         pending_lease_request = std::make_pair(nullptr, TaskID::Nil());
 
         if (status.ok()) {
-          if (reply.canceled()) {
+          if (reply.runtime_env_setup_failed()) {
+            // If the runtime_env failed to be set up, we fail all of the pending
+            // tasks in the queue. This makes an implicit assumption that runtime_env
```

Comment:
Just making sure I understand: the reason it's okay to fail all the tasks in the queue is that the runtime env hash is part of the scheduling key, so every task in the queue is trying to use the same runtime env, right?

Reply:
Yep, that's right.